.gitignore

```diff
@@ -1,9 +1,45 @@
-.vscode/
```

Comment:
Please keep .vscode and .idea, which are generated by vscode and clion respectively